File: src/main/java/com/ice/student/Student.java
package com.ice.student;

public class Student {
    String name;
    String id;
    int math;
    int english;
    static int pass = 60;

    public Student(String name, int english , int math) {
        this.name = name;
        this.english = english;
        this.math = math;
    }
    public int highest(){

//        int max =  ;
        /*if(english > math){
            max = english;
        }else{
            max = math;
        }*/
        return (english > math) ? english : math ;
    }
    public void print() {
        int average = getAverage();
        System.out.print(name + "\t" + english + "\t" + math + "\t" +
                getAverage() + "\t" +
                ((average >= pass) ? "PASS" : "FAILED"));
        char grading = 'F';
        switch (average/10){
            case 10:
            case 9:
                grading = 'A';
                break;
            case 8:
                grading = 'B';
                break;
            case 7:
                grading = 'C';
                break;
            case 6:
                grading = 'D';
                break;
            default:
                grading = 'F';
        }
        System.out.println("\t" + grading);
        /*if (getAverage() >= 90 && getAverage() <= 99){
            grading = 'A';
        }else if (getAverage() >= 80 && getAverage() <= 89){
            grading = 'B';
        }else if (getAverage() >= 70 && getAverage() <= 79){
            grading = 'C';
        }else if (getAverage() >= 60 && getAverage() <= 69){
            grading = 'D';
        }*/

        /*if (getAverage() >= 60){
            System.out.println("\tPASS");
        }else{
            System.out.println("\tFAILED");
        }*/
    }
    public int getAverage(){
        return (english + math)/2;
    }
}

File: src/main/kotlin/com/ice/student/Student.kt
package com.ice.kotlin

import java.util.*

fun main() {
//    userInput()
    Student.pass = 50
    val stu = Student("ice", 87, 69)
    val stu1 = Student("ice", 60, 40)
    val stu2 = Student("ice", 33, 34)
    stu.print()
    stu1.print()
    stu2.print()
    println("High score: ${stu.highest()}")
}
class Student (var name: String?, var english: Int, var math: Int){
    companion object{
        @JvmStatic
        var pass = 60
        fun test() {
            println("Testing")
        }
    }
    fun print(){
        println("$name\t$english\t$math\t${gatAverage()}\t${passOrFailed()}\t${grading()}")
    }
    fun grading() = when (gatAverage()){
        in 90..100 -> 'A'
        in 80..89 -> 'B'
        in 70..79 -> 'C'
        in 60..69 -> 'D'
        else -> 'F'
    }
    fun passOrFailed() = if (gatAverage() >= pass) "PASS" else "FAILED"
    fun gatAverage() = (english+math)/2

    fun highest() = if (english > math){
        println("english")
        english
    } else {
        println("math")
        math
    }

    fun nameCheck () = name?.length
}
private fun userInput() {
    val scanner = Scanner(System.`in`)
    print("Please enter student's name:")
//    var name = scanner.next()
    var name = null
    print("Please enter student's english:")
    var english = scanner.nextInt()
    print("Please enter student's math:")
    var math = scanner.nextInt()
}


